### ISSUE TITLE
ci: remote platform test

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-flow#3429]
Investigate using containerisation technology such as docker to create a "virtual remote platform" for use with Travis tests. This would enable CI for the following tests:

```console
$ git grep --name-only set_test_remote -- tests
tests/authentication/09-remote-suite-same-name.t
tests/authentication/10-remote-suite-passphrase-cache.t
tests/broadcast/09-remote.t
tests/cylc-cat-log/01-remote.t
tests/cylc-cat-log/05-remote-tail.t
tests/cylc-cat-log/08-editor-remote.t
tests/cylc-cat-log/09-cat-running.t
tests/cylc-cat-log/10-remote-no-retrieve.t
tests/cylc-cat-log/11-remote-retrieve.t
tests/cylc-kill/00-multi-hosts-compat.t
tests/cylc-kill/01-multi-hosts.t
tests/cylc-message/00-ssh.t
tests/cylc-poll/04-poll-multi-hosts.t
tests/database/03-remote.t
tests/events/10-task-event-job-logs-retrieve.t
tests/events/11-cycle-task-event-job-logs-retrieve.t
tests/events/15-host-task-event-handler-retry-globalcfg.t
tests/events/17-task-event-job-logs-retrieve-command.t
tests/events/32-task-event-job-logs-retrieve-2.t
tests/events/33-task-event-job-logs-retrieve-3.t
tests/host-select/00-simple.t
tests/job-kill/01-remote.t
tests/job-submission/02-job-nn-remote-host.t
tests/job-submission/07-multi.t
tests/job-submission/08-activity-log-host.t
tests/job-submission/09-activity-log-host-bad-submit.t
tests/job-submission/13-tidy-submits-of-prev-run-remote-host.t
tests/job-submission/17-remote-localtime.t
tests/lib/bash/test_header
tests/reload/19-remote-kill.t
tests/remote/00-basic.t
tests/restart/13-bad-job-host.t
tests/restart/26-remote-kill.t
```

Requirements:
* https://github.com/cylc/cylc-docker/issues/2
* Container should see a different filesystem.
* Container should have its own installation of Cylc Flow.

Nice To Have:
* Install batch systems e.g. PBS Pro (this is a bit of a faff but might be OK for a nightly CI build)

Leaving milestone blank for the moment while we prioritise, this functionality would be useful in the Cylc8 timeframe whilst we get deployment sorted out. Once https://github.com/cylc/cylc-docker/issues/2 is done this may turn out to be quick and easy...

**Comment by kinow** [comment]
Might be worth considering the move to GitHub actions if it's faster than Travis (I believe it's running on Azure/MS infra, which seems to be quite good looking at the Python repo), or if we can use their add-ons: [GitHub Marktplet Actions](https://github.com/marketplace?utf8=%E2%9C%93&type=actions&query=ssh).

**Comment by oliver-sanders** [comment]
Superseded by #3641
